lib/stat_tracker.rb

```diff
@@ -64,9 +64,90 @@ def average_goals_by_season
       total_goals_per_game = games_per_season.map do |game|  #very similar to total_goals_per_game in previus method
         game.home_goals + game.away_goals
       end
-      average = total_goals_per_game.sum / total_goals_per_game.length.to_f
+      average = total_goals_per_game.sum / total_goals_per_game.length.to_f # create module with average method??
       goals_by_season[season] = average.round(2)
       goals_by_season
     end
   end
+
+  # This only requires team information.
+  # It should probably move to team collection eventually.
+  def count_of_teams
+    @team_collection.teams.length
+  end
+
+  # uses both team and game_team info, needs to live in stat_tracker.
+  def best_offense
+    team_ids = @team_collection.all.map{|team| team.team_id}  # This could be shifted to use the game_team_collection data, just use a #uniq at the end
+
+    games_by_team = team_ids.reduce({}) do |games_by_team, team_id| # this snippet would better serve us in the game_team collection to be used by other methods
```

Comment:
In our game teams collection there is already an instance variable named games by team im not sure if that would cause any problems but just FYI.